src/metabase/api/field.clj

```diff
@@ -86,21 +154,45 @@
   [_ _]
   empty-field-values)
 
+(defn validate-human-readable-pairs
+  "Human readable values are optional, but if present they must be
+  present for each field value. Throws if invalid, returns a boolean
+  indicating whether human readable values were found."
+  [value-pairs]
+  (let [human-readable-missing? #(= ::not-found (get % 1 ::not-found))
+        has-human-readable-values? (not-any? human-readable-missing? value-pairs)]
+    (api/check (or has-human-readable-values?
+                   (every? human-readable-missing? value-pairs))
+      [400 "If remapped values are specified, they must be specified for all field values"])
+    has-human-readable-values?))
 
-;; TODO - not sure this is used anymore
-(api/defendpoint POST "/:id/value_map_update"
-  "Update the human-readable values for a `Field` whose special type is `category`/`city`/`state`/`country`
-   or whose base type is `type/Boolean`."
-  [id :as {{:keys [values_map]} :body}]
-  {values_map su/Map}
+(defn- update-field-values! [field-value-id value-pairs]
+  (let [human-readable-values? (validate-human-readable-pairs value-pairs)]
+    (api/check-500 (db/update! FieldValues field-value-id
+                     :values (map first value-pairs)
+                     :human_readable_values (when human-readable-values?
+                                              (map second value-pairs))))))
+
+(defn- create-field-values!
+  [field value-pairs]
+  (let [human-readable-values? (validate-human-readable-pairs value-pairs)]
+    (db/insert! FieldValues
+      :field_id (:id field)
```

Comment:
In cases like this we have a function in `util` called `get-id`. e.g. `(u/get-id field)`. There's a couple advantages here: 

it's guaranteed to return an integer ID, or it will throw an Exception along the lines of `not something with an ID: <whatever the arg was>`. So if `field` is somehow `nil` that would be a little easier to read than the SQL exception about violated `NOT NULL` constraints would be. Plus it would save us a failing DB call. In other places where we wouldn't be tripping a constraint it sort of serves as an assertion making sure we're actually getting a valid object, meaning you get the equivalent of a `:pre` precondition check for free

The other advantage is `get-id`is that works with either an object *or* ID: 

```clojure
(u/get-id {:id 1}) ; -> 1
(u/get-id 1)       ; -> 1
(u/get-id nil)     ; -> Exception!
```

That means you could write the function like

```clojure
(defn- create-field-values! [field-or-id value-pairs]
  ...
  (db/insert! FieldValues
    :field_id (u/get-id field-or-id)))
```

which gives you some extra flexibility.

Normally I'm weary of functions that do something we already had a perfectly fine way of doing (e.g., using `:id`) but I think the extra magic on offer justifies the increased cognitive load.

Either way no need to changes things here, I'm just letting you know it exists for future code if you see some places where using it makes more sense